tpe/plugin/src/SimulationFeatures_TEST.cc

```diff
@@ -0,0 +1,240 @@
+    auto sphereShape = sphereCollision->CastToSphereShape();
+    EXPECT_NEAR(1.0, sphereShape->GetRadius(), 1e-6);
+
+    auto sphere2 = sphereLink->AttachSphereShape(
+      "sphere2", 1.0, Eigen::Isometry3d::Identity());
+    EXPECT_EQ(2u, sphereLink->GetShapeCount());
+    EXPECT_EQ(sphere2, sphereLink->GetShape(1));
+  
+    auto ground = world->GetModel("box");
+    auto groundLink = ground->GetLink(0);
+    auto groundCollision = groundLink->GetShape(0);
+    auto boxShape = groundCollision->CastToBoxShape();
+    EXPECT_EQ(ignition::math::Vector3d(100, 100, 1),
+              ignition::math::eigen3::convert(boxShape->GetSize()));
+
+    auto box2 = groundLink->AttachBoxShape(
+      "box2",
+      ignition::math::eigen3::convert(
+        ignition::math::Vector3d(1.2, 1.2, 1.2)),
+      Eigen::Isometry3d::Identity());
+    EXPECT_EQ(2u, groundLink->GetShapeCount());
+    EXPECT_EQ(box2, groundLink->GetShape(1));
+
+    auto cylinder = world->GetModel("cylinder");
+    auto cylinderLink = cylinder->GetLink(0);
+    auto cylinderCollision = cylinderLink->GetShape(0);
+    auto cylinderShape = cylinderCollision->CastToCylinderShape();
+    EXPECT_NEAR(0.5, cylinderShape->GetRadius(), 1e-6);
+    EXPECT_NEAR(1.1, cylinderShape->GetHeight(), 1e-6);
+
+    auto cylinder2 = cylinderLink->AttachCylinderShape(
+      "cylinder2", 3.0, 4.0, Eigen::Isometry3d::Identity());
+    EXPECT_EQ(2u, cylinderLink->GetShapeCount());
+    EXPECT_EQ(cylinder2, cylinderLink->GetShape(1));
+
+    // Test the bounding boxes in the local frames
+    auto sphereAABB =
+      sphereCollision->GetAxisAlignedBoundingBox(*sphereCollision);
+
+    auto groundAABB =
+      groundCollision->GetAxisAlignedBoundingBox(*groundCollision);
+    
+    auto cylinderAABB =
+      cylinderCollision->GetAxisAlignedBoundingBox(*cylinderCollision);
+
+    EXPECT_EQ(ignition::math::Vector3d(-1, -1, -1),
+              ignition::math::eigen3::convert(sphereAABB).Min());
+    EXPECT_EQ(ignition::math::Vector3d(1, 1, 1),
+              ignition::math::eigen3::convert(sphereAABB).Max());
+    EXPECT_EQ(ignition::math::Vector3d(-50, -50, -0.5),
+              ignition::math::eigen3::convert(groundAABB).Min());
+    EXPECT_EQ(ignition::math::Vector3d(50, 50, 0.5),
+              ignition::math::eigen3::convert(groundAABB).Max());
+    EXPECT_EQ(ignition::math::Vector3d(-0.5, -0.5, -0.55),
+              ignition::math::eigen3::convert(cylinderAABB).Min());
+    EXPECT_EQ(ignition::math::Vector3d(0.5, 0.5, 0.55),
+              ignition::math::eigen3::convert(cylinderAABB).Max());
+  }
+}
+
+TEST_P(SimulationFeatures_TEST, FreeGroup)
+{
+  const std::string library = GetParam();
+  if (library.empty())
+    return;
+
+  auto worlds = LoadWorlds(library, TEST_WORLD_DIR "/shapes.world");
+
+  for (const auto &world : worlds)
+  {
+    auto model = world->GetModel("sphere");
+    auto freeGroup = model->FindFreeGroup();
+    EXPECT_NE(nullptr, freeGroup);
```

Comment:
This should be `ASSERT_NE`, so if the check fails, the test ends here instead of crashing on the line below.